wgpu-core/src/device/mod.rs

```diff
@@ -1094,12 +1094,12 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                             binding_model::BindingType::ReadonlyStorageBuffer => {
                                 (BIND_BUFFER_ALIGNMENT, wgt::BufferUsage::STORAGE_READ)
                             }
-                            binding_model::BindingType::Sampler
-                            | binding_model::BindingType::ComparisonSampler
-                            | binding_model::BindingType::SampledTexture
-                            | binding_model::BindingType::ReadonlyStorageTexture
-                            | binding_model::BindingType::WriteonlyStorageTexture => {
-                                panic!("Mismatched buffer binding for {:?}", decl)
+                            v @ binding_model::BindingType::Sampler
+                            | v @ binding_model::BindingType::ComparisonSampler
+                            | v @ binding_model::BindingType::SampledTexture
+                            | v @ binding_model::BindingType::ReadonlyStorageTexture
+                            | v @ binding_model::BindingType::WriteonlyStorageTexture => {
+                                panic!("Mismatched buffer binding type for {:?}. Expected a type of UniformBuffer, StorageBuffer or ReadonlyStorageBuffer but found {:?}", decl, v)
```

Comment:
can we just use `decl.ty` here instead of pattern binding `v @`?

Reply:
Fixed